src/core/server/http/integration_tests/lifecycle_handlers.test.ts

```diff
@@ -0,0 +1,197 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { resolve } from 'path';
+import * as kbnTestServer from '../../../../test_utils/kbn_server';
+import { clusterClientMock } from './core_service.test.mocks';
+
+const pkgPath = resolve(__dirname, '../../../../../package.json');
+const actualVersion = require(pkgPath).version;
+
+const versionHeader = 'kbn-version';
+const xsrfHeader = 'kbn-xsrf';
+const nameHeader = 'kbn-name';
+
+const whitelistedTestPath = '/xsrf/test/route/whitelisted';
+
+const kibanaName = 'my-kibana-name';
+
+describe('core lifecycle handlers', () => {
+  let root: ReturnType<typeof kbnTestServer.createRoot>;
+
+  beforeEach(async () => {
+    root = kbnTestServer.createRoot({
+      server: {
+        name: kibanaName,
+        customResponseHeaders: {
+          'some-header': 'some-value',
+        },
+        xsrf: { disableProtection: false, whitelist: [whitelistedTestPath] },
+      },
```

Comment:
I'm doing all three handlers integration tests in the same file, so the root creation has settings for all the 3 handlers.  Not sure if I should split/isolate per file